### PR TITLE
nixos/stunnel: Fix CA files

### PR DESCRIPTION
**Pull request by dasJ**
The default is the .crt file from the ssl package.
However, with CApath, it's assumed to be a directory containing
certificates, while with CAfile, it's properly recognised as a file.

Relevant strace output:
stat("/etc/ssl/certs/ca-certificates.crt/2e5ac55d.0", 0x7fc580f3d610) = -1 ENOTDIR (Not a directory)

cc @lschuermann

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by lschuermann** [comment]
Actually I have to admit that I never really used the client mode of `stunnel` in production with this module. If you're using it, maybe we should consider adding a meta maintainers field with us both as [outlined here](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/meta.nix). I think this just means adding a `meta.maintainers` attribute.

File: nixos/modules/services/networking/stunnel.nix [hunk 1]
```diff
@@ -195,7 +195,7 @@ in
                connect = ${v.connect}
                verifyChain = ${yesNo v.verifyChain}
                verifyPeer = ${yesNo v.verifyPeer}
-               ${optionalString (v.CAPath != null) "CApath = ${v.CAPath}"}
+               ${optionalString (v.CAPath != null) "CAfile = ${v.CAPath}"}
```
Comment (lschuermann) [inline nixos/modules/services/networking/stunnel.nix]: I think there is nothing wrong with this option per se, but just the default value that gets substituted. Maybe it's sufficient to change `${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt` to `${pkgs.cacert}/etc/ssl/certs`

**Comment by lschuermann** [comment]
Discussed the fixes in person at NixCon, similar to what I've proposed. This seems to be the right™ solution. Tested, working!

Thank you for discovering and fixing this, @dasJ! Really cool that we now have an official maintainer for the module's client side as well.

Ping @Infinisil, as soon as @dasJ confirms this to be working, it can be merged IMHO.

**Comment by dasJ** [comment]
Tested working

**Comment by lschuermann** [comment]
@GrahamcOfBorg eval

**Comment by chkno** [comment]
This sets `CAFile` (upper-case F), but the configuration option in [the manual](https://www.stunnel.org/manual.html) is `CAfile` (lower-case F).  So I'm confused how this change had any effect.

I carried over this misspelled `CAFile` in the module free-from refactoring in https://github.com/NixOS/nixpkgs/pull/152065 without noticing the problem.

I only just now noticed the problem when looking at one of my generated `/etc/stunnel.cfg` files and I see that it has both a `CAfile` line that I specified that's being used and an additional `CAFile` line from this default that I don't think does anything.

**Comment by lschuermann** [comment]
@chkno You mind sending a patch to fix this?

**Comment by chkno** [comment]
Oh, wait, my mistake: stunnel config directives are case-insensitive: All [option name comparisons are done with `strcasecmp`](https://fossies.org/linux/misc/legacy/stunnel-5.79.tar.gz/stunnel-5.79/src/options.c#l_1705).  So using `CAFile` instead of `CAfile` is fine as long as everyone agrees.

But if the same option is specified with different letter cases, instead of merging or conflicting, they're just both passed and then the one that happens to appear last in the config is the one that is used.  This is not great.  I'll prepare a PR to add an assertion against this.

**Pull request merged**